google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb

```diff
@@ -484,6 +490,72 @@ def dead_letter_max_delivery_attempts= new_dead_letter_max_delivery_attempts
           @resource_name = nil
         end
 
+        ##
+        # Removes an existing dead letter policy. A dead letter policy specifies the conditions for dead lettering
+        # messages in the subscription. If a dead letter policy is not set, dead lettering is disabled.
+        #
+        # See {#dead_letter_topic}, {#dead_letter_topic=}, {#dead_letter_max_delivery_attempts},
+        # {#dead_letter_max_delivery_attempts=} and {#dead_letter_policy?}.
+        #
+        # @return [Boolean] `true` if an existing dead letter policy was removed, `false` if no existing dead letter
+        #   policy was present.
+        #
+        # @example
+        #   require "google/cloud/pubsub"
+        #
+        #   pubsub = Google::Cloud::PubSub.new
+        #
+        #   sub = pubsub.subscription "my-topic-sub"
+        #
+        #   sub.dead_letter_topic.name #=> "projects/my-project/topics/my-dead-letter-topic"
+        #   sub.dead_letter_max_delivery_attempts #=> 10
+        #
+        #   sub.dead_letter_policy? #=> true
+        #   sub.remove_dead_letter_policy!
+        #   sub.dead_letter_policy? #=> false
+        #
+        #   sub.dead_letter_topic #=> nil
+        #   sub.dead_letter_max_delivery_attempts #=> nil
```

Comment:
I would group these differently: first group all the "before" state together, then show the call, and group all the "after" state together:

```ruby
        #   sub.dead_letter_policy? #=> true
        #   sub.dead_letter_topic.name #=> "projects/my-project/topics/my-dead-letter-topic"
        #   sub.dead_letter_max_delivery_attempts #=> 10
        #
        #   sub.remove_dead_letter_policy!
        #
        #   sub.dead_letter_policy? #=> false
        #   sub.dead_letter_topic #=> nil
        #   sub.dead_letter_max_delivery_attempts #=> nil
```

Reply:
Good call, I agree. Updated.